CMakeLists.txt

```diff
@@ -136,6 +163,12 @@ function(cprover_default_properties)
     set(CBMC_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY
             "Developer ID Application: Daniel Kroening")
 
+    foreach (target IN LISTS ARGN)
+        message(${target})
+        target_compile_options(${target} PRIVATE -Wall -Wpedantic -Werror -Wno-deprecated-declarations -Wswitch-enum)
```

Comment:
For other reviewers: This means we only set our warning/werror flags for targets that have `cprover_default_properties` called for them. We should do a check to make sure this is the case for all our own targets. Not setting werror stuff globally is so we can avoid having to patch dependencies just so we can avoid generating a warning.